### ISSUE TITLE
Invalid PAF files produced

### ISSUE THREAD

**Issue by kdm9** [waveygang/wfmash#273]
Hi folks, thanks for WFmash and associated tools.

I've found what I think is a bug producing invalid PAF output. 

When writing alignments, the main PAF fields are written only if `aln.ok` [here](https://github.com/waveygang/wfmash/blob/1f485f3d16c17ace8f4a623dc5add601538cae8a/src/common/wflign/src/wflign_patch.cpp#L2241), however [some extra fields are written additionally](https://github.com/waveygang/wfmash/blob/1f485f3d16c17ace8f4a623dc5add601538cae8a/src/common/wflign/src/wflign_patch.cpp#L2051-L2054), outside this function and regardless of the previous conditional. ~~This means that if `aln.ok` is false, you get invalid lines consisting of *only* the `pt` and `iv` tags written in lines 2051-2054.~~ Nope, this wasn't the cause, see #274  

Additionally, there is an extra tab there that seems superfluous.

Will send a PR to hopefully fix this soon.

best,
K

**Comment by kdm9** [comment]
an example of the invalid paf lines:

```
araport11#1#chr2        19698289        10000   75000   +       1k1g-6909#1#chr2        19787798        29820   105334  50652   89852   4       gi:f:0.999684   bi:f:0.563727   md:f:0.999035   cg:Z:544=990D2201=1X1032=1X18=1X41=14338I14916D10467=1X28=1D13413=1X3=1X2994=1X271=8943D899=2D1X5479=1X29=1X13233=
araport11#1#chr2        19698289        13839   28177   -       1k1g-6909#1#chr2        19787798        34649   48849   14194   14338   20      gi:f:0.999437   bi:f:0.989957   md:f:0.999035   cg:Z:1=2I2=1X1=2X1=136I1X584=1X5267=1X8338=             pt:Z:true               iv:Z:true
araport11#1#chr2        19698289        28150   28177   +       1k1g-6909#1#chr2        19787798        49538   49565   27      27      255     gi:f:1  bi:f:1  md:f:0.999035   cg:Z:27=                pt:Z:true               iv:Z:false
        pt:Z:true               iv:Z:false
araport11#1#chr2        19698289        28150   28177   +       dl20-8285#1#chr2        23046220        145624  145651  27      27      255     gi:f:1  bi:f:1  md:f:0.997073   cg:Z:27=                pt:Z:true               iv:Z:false
        pt:Z:true               iv:Z:false

```

**Comment by ASLeonard** [comment]
Strange, I sporadically saw this before in #263 but all the invalid lines I came across were fixed with #264. Still, guarding against writing the `pt`/`iv` tags if no line was written seems a good idea regardless even if the root cause is still lurking

**Comment by kdm9** [comment]
Agreed, I expected that would fix it but I compiled that from source and it's still there in current main. It *seems* that this is "fixed" in #274, though as you say the root cause might still be somewhere deep in the alignment logic that I'm not sufficiently familiar with to follow. If need be, I can supply the problematic chromosomes, the data is public-ish.

best,
K

**Comment by baozg** [comment]
Inversion patching try to use these records to represent small-scale inversion, but I think these may introduce more problem in the downstream graph building. Now they are represent as overlapping alignment with the forward alignment.
